### DISCUSSION TITLE
Please how to setup android service account

### DISCUSSION THREAD

**Discussion by monkhsaikhan** [imdhemy/laravel-in-app-purchases#381] [Show and tell]
I configured following user manual. But I cant Google Play Developer Account linked to Google Cloud Project. Because this feature removed from Google Play Console. Google play console > Set up > API access not shown. My account is owner. 
Then I create service account on Google Cloud console, generate key then download json. I configured on liap.php config file. On Google play console > Users and Permissions > Add my service account to users.
"The current user has insufficient permissions to perform the requested operation" error when verify google subscription purchase

I call like this -> `Subscription::googlePlay()->id('my_product_id')->token('my_purchase_token')->get()`

this function call this url with get method: 

`https://androidpublisher.googleapis.com/androidpublisher/v3/applications/{my_package_id}/purchases/subscriptions/{my_product_id}/tokens/{mytoken}`

please help me. What should I do?

**Comment by ghost** [comment]
I'm having the same issues, I've asked to Google Developer Support for that same endpoint and here is the answer they gave me:

I'm replying on behalf of {somebody}. I'm so sorry for the confusion this has caused you. To fix this, kindly make sure that your user/service account has sufficient [permission](https://support.google.com/googleplay/android-developer/answer/9844686) in the "Users and permissions" list for them to make an API request. 

For example, to use the Google Play Billing APIs, you must grant the following permissions:

 * View financial data, orders, and cancellation survey responses
 * Manage orders and subscriptions
Also, kindly make sure that you have configured OAuth and Service Accounts accordingly. For more information about this, you may visit this [documentation](https://developers.google.com/android-publisher/getting_started) from Google Play Developer API.

------------------------------------------------------------
I want to say at least in my case I wasn't able to find those permission in the step 2 When creating a service account, I think this permission are granted in another place...

**Reply by ghost** [reply]
Well I'm gonna kindly reply to myself with this Issue solved, as Google has been changing some things I've figured out how to add the permission I was talking earlier, they are in section Users and permission BUT in Google Play Console, NOT in Google Cloud Platform, in my case I was always looking for them in GCP, what I've done:

1 - Invite a new User and don't forget add the app this user will granted the access
![Screenshot 2024-02-14 095330](https://github.com/imdhemy/laravel-in-app-purchases/assets/159159420/c5f185b0-730c-47da-9302-7bf5c2a2e722)
2 - In Account permissions grant access to the both inside Financial data
![Screenshot 2024-02-14 095557](https://github.com/imdhemy/laravel-in-app-purchases/assets/159159420/6adfd44c-6f27-48b6-93ba-938d0f61e54a)

After finishing, get back again to Account Permissions cause in my case the permission were unchecked, then if check both permission again and save.. I have created another subscription after granting all this permissions but this might be not necessary...

**Comment by mattli001** [comment]
Recently, I have been using packages to integrate Google Play Console's IAP. Here are two detailed configuration documents for reference:
1. [Google Play Service Account Key](https://documentation.qonversion.io/docs/service-account-key-android#3-link-google-cloud-project) (last updated 2024-05)
2. [Android In-App Products Setup](https://documentation.qonversion.io/docs/android-in-app-products) (last updated 2024-01)
